README.rst

```diff
@@ -334,10 +334,119 @@ This recipe is a little different from other recipes in that it's just a script
    COPY --from=rocky /usr/local /usr/local
    # Only needs to be run once for all recipes
    RUN for patch in /usr/local/share/just/container_build_patch/*; do "${patch}"; done
-   
+
    RUN dnf install -y --enablerepo=rocky-appstream telnet # This line is just an example
    ...
 
+CUDA
+----
+
+=========================== ============
+Name                        CUDA
+Build Args                  ``CUDA_REPO_REF`` - The version of the CUDA container repo parsed
+Build Args                  ``CUDA_VERSION`` - Version of CUDA to install (e.g. ``10.2`` or ``11.0.7``)
+Build Args                  ``CUDNN_VERSION`` - Optional: Version of CUDNN to install. (e.g. ``7`` or ``8``)
+Build Args                  ``CUDA_RECIPE_TARGET`` - Optional: Sets the target level of CUDA install (explained below). Default: ``runtime``
+Environment Variable        ``NVIDIA_VISIBLE_DEVICES`` - Required: Sets while nvidia devices to connect. Set to ``all``
+Environment Variable        ``NVIDIA_DRIVER_CAPABILITIES`` - Optional: Default is ``compute,utility`` when unset
+Environment Variable        ``NVIDIA_REQUIRE_*`` - Optional: Sets test conditions to prevent running on incompatible systems
+Output dir                  ``/usr/local``
+Minimum Dockerfile frontend docker/dockerfile:1.3-labs or docker/dockerfile:1.4
+=========================== ============
+
+While starting from a base image with CUDA already setup for docker is ideal, when we have to be based on a specific image (e.g. hardened images), this becomes impossible or impractical. Instead we need to start with a particular image and add CUDA support to it.
+
+Currently only RHEL and Ubuntu based images are supported (not Fedora or Debian).
+
+There are many steps to setting up CUDA in an image:
+
+- Setting up CUDA repo and GPG key
+- Installing the right packages, so that we limit the amount of bloat to the image
+- Setting various environment variables at container run time
+- Setting certain environment variables at container create time
```

Comment:
what is the difference between these to cases? should this say 'container build time'? (mentioned in the next sections too)